### ISSUE TITLE
Make configury aware of git snapshot versions

### ISSUE THREAD

**Issue by patrickmacarthur** [ofiwg/libfabric#1017]
(Creating a separate issue rather than hijacking the existing PR)

If I understand the issue correctly, PR #993 is really trying to solve the problem that the configury doesn't know about the git version, and thus generates a spec file with the wrong version unless it is massaged appropriately.

I propose that we do what I suggested a while back (and what EXS and openmpi do), and modify `configure.ac` to run a script to detect the version (via the `git describe` output or a `.version` file in the top level directory) rather than hardcoding the value.  This would avoid the need to patch `configure.ac` in the nightly script, and avoid the need to patch the version in the spec file.  I can code up a separate PR in the next couple of days to do this.

@shefty @jsquyres @pmmccorm Thoughts on this solution?

**Comment by jsquyres** [comment]
I also suggested a solution like this a long time ago, and people hated it for some reason (I don't recall why).  

FWIW: Open MPI does a similar thing; it reads a VERSION file during autogen time to effectively substitute in the version into configure.

But your solution sounds a little more comprehensive: it chooses between `.version` and `git describe`, though -- I'd be interested in seeing that.

**Comment by pmmccorm** [comment]
I think this could be good and solve most problems. It would complement #993 

And yeah the tricky part would be when to choose between ".version" and "git describe" that works for everyone. I'd be interested in seeing it too.

**Comment by shefty** [comment]
Is this issue still valid?  What needs to be done here?

**Comment by pmmccorm** [comment]
I think this isn't needed anymore with the addition of the file exclusion mechanism in distscript.pl

**Comment by patrickmacarthur** [comment]
I think it may still be worthwhile in that it may eliminate some of the hacks that people are currently using to get around the issue, but I currently lack the time to implement it.

@jsquyres I looked again at what I did for EXS and I had the autogen.sh script set up the .version file based on "git describe" so that it was there when autoreconf was run, and I used a "post-checkout" hook to update the `.version` file when I changed branches (I didn't update it on every commit since that would cause a full rebuild on every commit).  So what I had there wasn't a true solution.
